app/serializers/current_user_serializer.rb

```diff
@@ -153,9 +157,23 @@ def custom_fields
   end
 
   def muted_category_ids
-    @muted_category_ids ||= CategoryUser.where(user_id: object.id,
-                                               notification_level: TopicUser.notification_levels[:muted])
-      .pluck(:category_id)
+    CategoryUser.lookup(object, :muted).pluck(:category_id)
+  end
+
+  def tracked_category_ids
+    CategoryUser.lookup(object, :tracking).pluck(:category_id)
+  end
+
+  def watched_category_ids
+    CategoryUser.lookup(object, :watching).pluck(:category_id)
+  end
+
+  def watched_first_post_category_ids
+    CategoryUser.lookup(object, :watching_first_post).pluck(:category_id)
+  end
+
+  def top_category_ids
+    UserSummary.new(object, Guardian.new).top_categories.pluck(:id)
```

Comment:
how expensive is this query? Just curious....

Reply:
also... instead of 4 queries to look up category user data here, just do one

Reply:
might as well add this to the serializer test as well

Reply:
@SamSaffron it may bit costly. `top_categories` method calling db 4 times with select queries _(I hope it can be reduced to 3)_. Since we are using this in two places I like to add `topic_count` and `post_count` columns to `category_users` table if possible.

And that method called by every user summary page visit too.

Reply:
why cant this just be done in 1 query?